### PR TITLE
Add chess support to aie2xclbin

### PR DESCRIPTION
**Pull request by spaceotter**
This copies the pile of hacks from aiecc.py to use the Vitis chess compiler with aie2xclbin (thereby allowing IREE to use chess).

### CODE REVIEW

File: tools/aie2xclbin/XCLBinGen.cpp [hunk 1]
```diff
@@ -501,6 +502,184 @@ static LogicalResult generateXCLBin(MLIRContext *context, ModuleOp moduleOp,
   return success();
 }
 
+static std::string chesshack(const std::string &input) {
```
Comment (stephenneuendorffer) [inline tools/aie2xclbin/XCLBinGen.cpp:505]: Can you factor this out into a library that can be called from aiecc.py?

Comment (spaceotter) [reply tools/aie2xclbin/XCLBinGen.cpp:505]: Just this one function or the whole thing?

Comment (stephenneuendorffer) [reply tools/aie2xclbin/XCLBinGen.cpp:555]: We shouldn't have the same code duplicated between aiecc.py and aie2xclbin.  This function seems like a good granularity to reduce that duplication?  I think the same goes for much of the other code you've factored out.

File: tools/aie2xclbin/XCLBinGen.cpp [hunk 2]
```diff
@@ -197,55 +198,113 @@ static LogicalResult generateCoreElfFiles(ModuleOp moduleOp,
...
+                                     "-d",
+                                     "+l",
+                                     std::string(bcfPath),
+                                     "-o",
+                                     std::string(elfFile),
+                                     "-f",
+                                     std::string(objFile)};
+      for (const auto &inc : extractedIncludes) {
+        flags.push_back(inc);
+      }
```
Comment (makslevental) [inline tools/aie2xclbin/XCLBinGen.cpp:258]: nit: can you remove braces for all single line for/ifs (ie follow upstream conventions...)

**Review by stephenneuendorffer — APPROVED** [review]

File: tools/aie2xclbin/XCLBinGen.cpp (hunk 2 above)
Comment (makslevental) [reply tools/aie2xclbin/XCLBinGen.cpp]: @erieaton-amd thanks :)

**Pull request merged**